# Genome annotation

## Genome preparation

The genome preparation phase of the Ensembl gene annotation system involves loading an assembly into an Ensembl core database schema and then running a series of analyses on the loaded assembly to identify an initial set of genomic features.

The most important aspect of this phase is identifying repeat features as soft masking of the genome is used extensively later in the annotation process.
### Repeat Finding

After the genomic sequence is loaded into a database, it is screened for sequence patterns, including repeats, using RepeatMasker [1], (version 4.0.5 with parameters -nolow -engine "RMBlast"), dustmasker ([Morgulis, A., *et al.*, 2006](https://doi.org/10.1089/cmb.2006.13.1028)) and TRF ([Benson, G., 1999](https://doi.org/10.1093/nar/27.2.573)). In addition to the Repbase ([Bao, W., *et al*, 2015](https://doi.org/10.1186/s13100-015-0041-9)) library, where available, a custom repeat library is used with RepeatMasker. Custom libraries are created using RepeatModeler2 ([Flynn, J.M., 2020](https://doi.org/10.1073/pnas.1921046117)) and can be downloaded via the [Ensembl FTP](https://ftp.ebi.ac.uk/pub/databases/ensembl/repeats/unfiltered_repeatmodeler/species/).
### Low Complexity Feature Prediction

Transcription start sites are predicted using Eponine–scan [6], CpG islands longer than 400 bases are predicted using CpG ([Larsen, F., *et al*, 1992](https://doi.org/10.1016/0888-7543(92)90024-m)) and tRNAs are predicted using tRNAscan-SE [Chan, P.P., *et al*, 2019](https://doi.org/10.1007/978-1-4939-9173-0_1). The results of Eponine-scan, CpG, and tRNAscan are for display purposes only; they are not used in the gene annotation process.
## Generating Automatic, Evidence-based Protein-coding Gene Models

Genome annotation is generated primarily through alignment of publicly available transcriptomic data to the genome. Gaps in the annotation are filled via protein-to-genome alignments of a select set of proteins. The data and techniques employed to generate gene models are outlined here.
### Model Generation Through Alignment of Transcriptomic Data to the Genome

#### Aligning RNASeq Data
Where available, RNA-seq data are downloaded from [ENA](https://www.ebi.ac.uk/ena/) and utilised for annotation. The obtained reads are aligned to the genome using STAR ([Dobin A., *et al*, 2013](https://doi.org/10.1093/bioinformatics/bts635)) and models assembled using Scallop ([Shao M., Kingsford C., 2017](https://doi.org/10.1038/nbt.4020)).

#### Aligning Long-read Transcriptomic Data
Where available, long-read transcriptomic data (PacBio IsoSeq or Oxford Nanopore) are downloaded from [ENA](https://www.ebi.ac.uk/ena/) and used in the annotation process. The long-read data are mapped to the genome using Minimap2 ([Li H., 2018](https://doi.org/10.1093/bioinformatics/bty191)) with the recommended settings for Iso-Seq and Nanopore data.

#### Validating Protein-coding Models
Protein-coding models are validated by aligning the longest ORF against a database of eukaryotic UniProt proteins using Diamond ([Buchfink B., *et al.*, 2021](https://doi.org/10.1038/s41592-021-01101-x)). If the alignments are insufficient, additional validation is performed using RNASamba ([Camargo A.P., *et al*, 2020](https://doi.org/10.1093/nargab/lqz024)) and CPC2 ([Kang Y., *et al.*, 2017](https://doi.org/10.1093/nar/gkx428)).
### Model Generation Through Protein-to-Genome Alignments

Protein sequences are downloaded from public databases and aligned to the genome in a splice-aware manner using GenBlast ([She, R., *et al*, 2011](https://doi.org/10.1093/bioinformatics/btr342)). The proteins aligned to the genome are selected subsets of [UniProt](https://www.uniprot.org/) ([The UniProt Consortium, 2017](https://doi.org/10.1093/nar/gkw1099)) and [OrthoDB](https://www.orthodb.org/) ([Kriventseva E.K., *et al*, 2019](https://doi.org/10.1093/nar/gky1053)) proteins, chosen to provide broad and targeted coverage of the species' proteome of interest.

For GenBlast, a cut-off of 50% coverage and 30% identity, with an e-value threshold of e-1, is applied, and the exon repair option is enabled. The top 10 transcript models generated by GenBlast for each protein meeting these cut-offs are retained.

### Model Generation Through Mapping of Reference Annotations to the Genome

A whole genome alignment against either the human or Mouse GENCODE reference assembly is generated using LastZ ([Harris, R.S., 2007](https://www.bx.psu.edu/~rsharris/rsharris_phd_thesis_2007.pdf)). Syntenic regions are identified from this alignment and protein-coding annotations from the most recent released gene set is mapped to the genome through localised alignments using CESAR 2.0 ([Sharma, V.P., *et al*, 2017](https://doi.org/10.1093/bioinformatics/btx527)).
## Filtering Gene Models and Finalisation of Gene Set

### Prioritising models at each locus

Low quality models are removed, and data is collapsed and consolidated into a final set of gene models plus their associated non-redundant transcripts. When collapsing the data, priority is given to models derived from transcriptomic data; where RNA-seq data is fragmented or missing, homology data takes precedence.

### Addition of UTR to coding models
The set of coding models is extended into the untranslated regions (UTRs) using RNA-seq data (if available). The criterion for adding UTR from RNA-seq alignments to protein models lacking UTR (such as the protein-to-genome alignment models) is that the intron coordinates from the model missing UTR exactly match a subset of the coordinates from the UTR donor model.

### Stable identifiers
Stable identifiers are assigned to each gene, transcript, exon and translation. When annotating a species for the first time, these identifiers are auto-generated. In all subsequent annotations for a species, the stable identifiers are propagated based on comparison of the new gene set to the previous gene set.

## Creating the Final Gene Set

### Identifying Small Non-coding RNAs

Small structured non-coding genes from [RFAM](https://rfam.org/) ([Nawrocki, E.P., *et al.*, 2015](https://doi.org/10.1093/nar/gku1063)) and [miRBase](https://www.mirbase.org/) ([Griffiths-Jones, S., *et al.*, 2006](https://doi.org/10.1093/nar/gkj112)) are analyzed using [NCBI-BLAST](https://blast.ncbi.nlm.nih.gov/Blast.cgi) ([Altschul, S.F., et al.,1990](https://doi.org/10.1016/s0022-2836(05)80360-2)), and models are generated using the Infernal software suite ([Nawrocki, E.P., and S.R. Eddy, 2013](https://doi.org/10.1093/bioinformatics/btt509)).
### Identifying Pseudogenes

Models with hits to known proteins that also display abnormalities such as absence of a start codon, non-canonical splicing, unusually small intron structures (< 75bp) or excessive repeat coverage, are reclassified as pseudogenes. Single-exon models with a corresponding multi-exon copy elsewhere in the genome are classified as processed (retrotransposed) pseudogenes.
### Identifying Long Non-coding RNAs

If a model fails to meet the criteria of any of the previously described categories, does not overlap a protein-coding gene, and has been constructed from transcriptomic data then it is considered as a potential lncRNA. Potential lncRNAs are additionally filtered to remove single-exon loci due to the unreliability of such models.
## Appendix - Further information
The Ensembl gene set is generated automatically, meaning that gene models are annotated using the Ensembl gene annotation pipelines. The main focus of the pipelines is to generate a conservative set of protein-coding gene models.

Every gene model produced by the Ensembl gene annotation pipeline is supported by biological sequence evidence (see the “Supporting evidence” link on the left-hand menu of a Gene page or Transcript page); ab initio models are not included in our gene set.