### Model Generation Through Protein-to-Genome Alignments

Protein sequences are downloaded from public databases and aligned to the genome in a splice-aware manner using GenBlast ([She, R., *et al*, 2011](https://doi.org/10.1093/bioinformatics/btr342)). The proteins aligned to the genome are selected subsets of [UniProt](https://www.uniprot.org/) ([The UniProt Consortium, 2017](https://doi.org/10.1093/nar/gkw1099)) and [OrthoDB](https://www.orthodb.org/) ([Kriventseva E.K., *et al*, 2019](https://doi.org/10.1093/nar/gky1053)) proteins, chosen to provide broad and targeted coverage of the species' proteome of interest.

For GenBlast, a cut-off of 50% coverage and 30% identity, with an e-value threshold of e-1, is applied, and the exon repair option is enabled. The top 10 transcript models generated by GenBlast for each protein meeting these cut-offs are retained.

